<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VisualizerWindowClass</class>
 <widget class="QMainWindow" name="VisualizerWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>641</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>776</width>
    <height>583</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>VisualizerWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="maximumSize">
       <size>
        <width>210</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>208</width>
         <height>568</height>
        </rect>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>10</y>
          <width>201</width>
          <height>541</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QGroupBox" name="displaygroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>100</height>
            </size>
           </property>
           <property name="title">
            <string>Display</string>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>149</width>
              <height>70</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QCheckBox" name="cb_displayPointcloud">
               <property name="text">
                <string>Pointcloud</string>
               </property>
              </widget>
             </item>
              <item>
                <widget class="QCheckBox" name="cb_displayGMPositions">
                  <property name="text">
                    <string>GM Positions</string>
                  </property>
                </widget>
              </item>
             <item>
              <widget class="QCheckBox" name="cb_displayEllipsoids">
               <property name="text">
                <string>Ellipsoids</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="cb_displayDensity">
               <property name="text">
                <string>Density</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="ellrendermodegroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>120</height>
            </size>
           </property>
           <property name="title">
            <string>Render Modes</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>160</width>
              <height>86</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string> Ellipsoids:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="co_ellrendermode">
               <item>
                <property name="text">
                 <string>Uniform color</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Color by weight</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Color by amplitude</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string> Density:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="co_densrendermode">
               <property name="currentText">
                <string>Fast (Projection)</string>
               </property>
               <property name="currentIndex">
                <number>2</number>
               </property>
               <item>
                <property name="text">
                 <string>Exact</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Fast (Octree)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Fast (Projection)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Sampling</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="elcolorgroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>100</height>
            </size>
           </property>
           <property name="title">
            <string>Ellipsoid Coloring</string>
           </property>
           <widget class="QWidget" name="layoutWidget_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>165</width>
              <height>74</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="spin_ellmin">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="decimals">
                <number>10</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>9999999999999999635896294965248.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.000500000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Min:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="spin_ellmax">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="wrapping">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="decimals">
                <number>10</number>
               </property>
               <property name="maximum">
                <double>9999999999999999635896294965248.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.000500000000000</double>
               </property>
               <property name="value">
                <double>0.050000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Max:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="co_ellrangemode">
               <property name="currentIndex">
                <number>1</number>
               </property>
               <item>
                <property name="text">
                 <string>Manually</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Auto (MinMax)</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Auto (MedMed)</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="coloringgroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>130</height>
            </size>
           </property>
           <property name="title">
            <string>Density Coloring (x100)</string>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>191</width>
              <height>76</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Min:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Max:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="spin_dscalemax">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="wrapping">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="decimals">
                <number>10</number>
               </property>
               <property name="maximum">
                <double>9999999999999999635896294965248.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.000500000000000</double>
               </property>
               <property name="value">
                <double>0.050000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="spin_dscalemin">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="decimals">
                <number>10</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>9999999999999999635896294965248.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.000500000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="cb_dscaleauto">
               <property name="text">
                <string>Auto</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSlider" name="sl_dscalepercentage">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>90</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="accelerationgroup">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>80</height>
            </size>
           </property>
           <property name="title">
            <string>Acceleration</string>
           </property>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>191</width>
              <height>45</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="spin_accthreshold">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="decimals">
                <number>10</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>9999999999999999635896294965248.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.000005000000000</double>
               </property>
               <property name="value">
                <double>0.000005000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Threshold(x100):</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QCheckBox" name="cb_accthauto">
               <property name="text">
                <string>Auto</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="gmvis::ui::DisplayWidget" name="openGLWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="loadPointcloudAction"/>
    <addaction name="loadMixtureModelAction"/>
    <addaction name="loadPureMixtureAction"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="loadPointcloudAction">
   <property name="text">
    <string>Load Pointcloud</string>
   </property>
  </action>
  <action name="loadMixtureModelAction">
   <property name="text">
    <string>Load Mixture Model</string>
   </property>
  </action>
   <action name="loadPureMixtureAction">
     <property name="text">
       <string>Load Mixture</string>
     </property>"
   </action>"
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DisplayWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>gmvis/ui/DisplayWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="VisualizerWindow.qrc"/>
 </resources>
 <connections/>
</ui>
